cse-cc-import-runner-app/src/main/java/com/farao_community/farao/cse/import_runner/app/util/GenericThreadLauncher.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package com.farao_community.farao.cse.import_runner.app.util;
+
+import com.farao_community.farao.cse.runner.api.exception.CseInternalException;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import com.farao_community.farao.cse.runner.api.resource.ThreadLauncherResult;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class GenericThreadLauncher<T, U> extends Thread {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(GenericThreadLauncher.class);
+    private final T threadable;
+    private final Method run;
+    private final Object[] args;
+
+    private ThreadLauncherResult<U> result;
+
+    public GenericThreadLauncher(T threadable, String id, Object... args) {
+        super(id);
+        this.run = getMethodAnnotatedWith(threadable.getClass());
+        this.threadable = threadable;
+        this.args = args;
+    }
+
+    @Override
+    public void run() {
+        try {
+            U threadResult = (U) this.run.invoke(threadable, args);
+            this.result = new ThreadLauncherResult<>(Optional.ofNullable(threadResult), false, null);
+        } catch (IllegalAccessException | InvocationTargetException e) {
+            LOGGER.error("Error occurred during CSE run", e);
+            if (checkInterruption(e)) {
+                //an interruption is not considered as an error because it is intentional
+                this.result = new ThreadLauncherResult<>(Optional.empty(), false, e);
```

Comment:
You don't even need to pass the error here. Maybe you can create static methods to build the object, ThreadLauncherResult.success(U) -- and U could be null -- that sets error to false and null to the exception. And another one ThreadLauncherResult.error(Exception) with just the exception and sets Optional.empty() and true by default.

Reply:
nope, this is not possible to have static methods with generic variable

java: non-static type variable U cannot be referenced from a static context